messaging/src/main/java/org/axonframework/queryhandling/SimpleQueryUpdateEmitter.java

```diff
@@ -63,6 +64,9 @@
     private final List<MessageDispatchInterceptor<? super SubscriptionQueryUpdateMessage<?>>> dispatchInterceptors =
             new CopyOnWriteArrayList<>();
 
+    private ConcurrentMap<SubscriptionQueryMessage<?, ?, ?>, Registration> registrations =
```

Comment:
Let's make the key of this map the identifier of the message instead of the entire message.
Small safer, but doesn't hurt I think. Added, we can change it back to the entire message if this proofs necessary later on.